lib/ansible/modules/cloud/amazon/ec2_placement_group_facts.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python
+#
+# This is a free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ec2_placement_group_facts
+short_description: List EC2 Placement Group(s) details
+description:
+    - List details of EC2 Placement Group(s).
+version_added: "2.5"
+author: "Brad Macpherson (@iiibrad)"
+options:
+  names:
+    description:
+      - A list of names to filter on. If a listed group does not exist, there
+        will be no corresponding entry in the result; no error will be raised.
+    required: false
+    default: []
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details or the AWS region,
+# see the AWS Guide for details.
+
+# List all placement groups.
+- ec2_placement_group_facts:
+  register: all_ec2_placement_groups
+
+# List two placement groups.
+- ec2_placement_group_facts:
+  names:
+     - my-cluster
+     - my-other-cluster
+  register: specific_ec2_placement_groups
+
+
+'''
+
+
+RETURN = '''
+placement_groups:
+  description: Placement group attributes
+  returned: always
+  type: complex
+  contains:
+    name:
+      description: PG name
+      type: string
+      sample: my-cluster
+    state:
+      description: PG state
+      type: string
+      sample: "available"
+    strategy:
+      description: PG strategy
+      type: string
+      sample: "cluster"
+
+'''
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (AnsibleAWSError,
+                                      connect_to_aws,
+                                      boto3_conn,
+                                      ec2_argument_spec,
+                                      get_aws_connection_info,
+                                      get_ec2_security_group_ids_from_names)
+
+
+def get_placement_groups_details(connection, module):
+    names = module.params.get("names")
+    if len(names) > 0:
+        response = connection.describe_placement_groups(
```

Comment:
All AWS API calls should have exception handling

Reply:
Fixed; one try block for both variants of the call.